licenses/LICENSE

```diff
@@ -0,0 +1,202 @@
+
+                                 Apache License
+                           Version 2.0, January 2004
+                        http://www.apache.org/licenses/
+
+   TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION
+
+   1. Definitions.
+
+      "License" shall mean the terms and conditions for use, reproduction,
```

Comment:
Should this folder be called binary-licenses or something to make it more clear that these are the license/notice that we use when assembling the relevant jars?

Reply:
I'll double check what other projects do, maybe there's a practice here that I'm missing.

Reply:
I guess there's no definite practice? 

Spark calls https://github.com/apache/spark/tree/master/licenses-binary
But Flink just puts them under licenses https://github.com/apache/flink/tree/master/licenses 

My stance is if there's a known practice we should just follow that, if there's not, I feel like it makes sense to just call it binary-licenses or something like that? cc @rdblue @danielcweeks

Reply:
Yes, there's no unique way to do that 😄 
For "clarity", I think that `binary-licenses` is better, I will rename.

Reply:
I did the change to `licenses-binary` folder (for clarity).